src/ex4/server/clients/pokemon_client.js

```
@@ -0,0 +1,38 @@
+// The Pokemon Client (using axios) goes here
+
+import axios from "axios";
+import { popularPokemons } from "./popular-pokemons.js";
+
+export class PokemonClient {
+  constructor() {
+    this.API_BASE = 'https://pokeapi.co/api/v2/pokemon';
+  }
+
+  async fetchPokemon(pokemonText) {
+    try {
+      const pokemons = pokemonText.split(',').map( el => el.trim() );
+      const promises = pokemons.map(pokemon => {
+        return axios.get(`${this.API_BASE}/${pokemon}/`);
+      });
+      const responses = await Promise.all(promises);
+      const elements = await Promise.all(responses.map(response => response.data));
+      return elements.map(element => {
+        return {
+          name: element.forms[0].name,
+          type: element.types[0].type.name,
+          id: element.id
+        }
+      });
+    } catch (error) {
+      this.handleFailure(error, pokemonText);
+    }
+  }
+
+  handleFailure(error, pokemon) {
```

so in this case you are 'swallowing' the error since you do not rethrow and nothing will happen in this case, was this intentional?

---

This fetchPokemon function is called from main.js and if it does not return the right data (and just logs this error) it is caught in main and "Failed to fetch" is added instead of pokemon.

---

so, yes, it was intentional